### PR TITLE
Update integration tests to use REST APIs in master branch (IV)

### PR DESCRIPTION
**Pull request by ashanthamara**
### saml
- AbstractSAMLSSOTestCase
- SAMLIdPInitiatedSSOTestCase
- SAMLInvalidIssuerTestCase
- SAMLQueryProfileTestCase
- SAMLSSOConsentTestCase

## Added following rest clients
- `KeystoreMgtRestClient`

## Related PRs
- https://github.com/wso2/product-is/pull/16223
- https://github.com/wso2/product-is/pull/16271
- https://github.com/wso2/product-is/pull/16354

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/KeystoreMgtRestClient.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    /**
+     * Check whether a certain certificate is already added in the tenant keystore.
+     *
+     * @param alias alias.
+     */
+    public Boolean checkCertInStore(String alias) throws Exception {
+        String endPointUrl = serverUrl + String.format(KEYSTORE_BASE_PATH, tenantDomain) + PATH_SEPARATOR + alias;
+
+        try (CloseableHttpResponse response = getResponseOfHttpGet(endPointUrl, getHeaders())) {
+            if (response.getStatusLine().getStatusCode() == HttpServletResponse.SC_OK) {
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/KeystoreMgtRestClient.java:73]: We can simplify this logic.

Comment (ashanthamara) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/KeystoreMgtRestClient.java]: Fixed with https://github.com/wso2/product-is/pull/16455/commits/c3f4848cd73fedff4c23504a9de3f89ea05a6eb7

**Review by madurangasiriwardena — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/KeystoreMgtRestClient.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        this.username = tenantInfo.getContextUser().getUserName();
+        this.password = tenantInfo.getContextUser().getPassword();
+    }
+
+    /**
+     * Upload the certificate to the tenant keystore. This API is not supported for super tenant.
+     *
+     * @param certificateRequest Certificate request object.
+     */
+    public void importCertToStore(CertificateRequest certificateRequest) throws Exception {
```
Comment (Thisara-Welmilla) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/KeystoreMgtRestClient.java:54]: Can throw the specific exception instead super class?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/OAuth2RestClient.java [hunk 3]
```diff
@@ -187,6 +188,21 @@ public void updateInboundDetailsOfApplication(String appId, Object inboundConfig
         }
     }
 
+    /**
+     * Delete an Inbound Configuration
+     *
+     * @param appId Application id.
+     * @param inboundType Inbound Type to be deleted.
+     */
+    public Boolean deleteInboundConfiguration(String appId, String inboundType) throws IOException {
```
Comment (Thisara-Welmilla) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/OAuth2RestClient.java:197]: In the method description add the @throws IOException <root/reason the exception from the POV of this method>. Let's keep this as a practice.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/KeystoreMgtRestClient.java (hunk 2 above)
Comment (ashanthamara) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/KeystoreMgtRestClient.java:54]: Fixed with https://github.com/wso2/product-is/pull/20096

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/OAuth2RestClient.java (hunk 3 above)
Comment (ashanthamara) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/OAuth2RestClient.java:197]: Fixed with https://github.com/wso2/product-is/pull/20096

**Pull request merged**